plugins/eddn.py

```diff
@@ -35,121 +46,147 @@
         self.replayfile.truncate()
         for line in self.replaylog:
-            self.replayfile.write('%s\n' % line)
+            self.replayfile.write(f'{line}\n')
+
         self.replayfile.flush()
 
     def close(self):
         if self.replayfile:
             self.replayfile.close()
+
         self.replayfile = None
 
-    def send(self, cmdr, msg):
-        uploaderID = cmdr
+    def send(self, cmdr: str, msg: Mapping[str, Any]):
+        uploader_id = cmdr
 
-        msg = OrderedDict([
+        to_send: OrderedDictT[str, str] = OrderedDict([
             ('$schemaRef', msg['$schemaRef']),
-            ('header',     OrderedDict([
-                ('softwareName',    '%s [%s]' % (applongname, sys.platform=='darwin' and "Mac OS" or system())),
+            ('header', OrderedDict([
+                ('softwareName',    f'{applongname} [{system() if sys.platform != "darwin" else "Mac OS"}]'),
                 ('softwareVersion', appversion),
-                ('uploaderID',      uploaderID),
+                ('uploaderID',      uploader_id),
             ])),
-            ('message',    msg['message']),
+            ('message', msg['message']),
         ])
 
-        r = self.session.post(self.UPLOAD, data=json.dumps(msg), timeout=self.TIMEOUT)
+        r = self.session.post(self.UPLOAD, data=json.dumps(to_send), timeout=self.TIMEOUT)
         if r.status_code != requests.codes.ok:
             logger.debug(f':\nStatus\t{r.status_code}URL\t{r.url}Headers\t{r.headers}Content:\n{r.text}')
+
         r.raise_for_status()
 
     def sendreplay(self):
         if not self.replayfile:
             return  # Probably closing app
 
-        status = self.parent.children['status']
+        status: Dict[str, Any] = self.parent.children['status']
 
         if not self.replaylog:
             status['text'] = ''
             return
 
+        localized: str = _('Sending data to EDDN...')
         if len(self.replaylog) == 1:
-            status['text'] = _('Sending data to EDDN...')
+            status['text'] = localized
+
         else:
-            status['text'] = '%s [%d]' % (_('Sending data to EDDN...').replace('...',''), len(self.replaylog))
+            status['text'] = f'{localized.replace("...", "")} [{len(self.replaylog)}]'
+
         self.parent.update_idletasks()
+
         try:
             cmdr, msg = json.loads(self.replaylog[0], object_pairs_hook=OrderedDict)
+
         except json.JSONDecodeError as e:
             # Couldn't decode - shouldn't happen!
             logger.debug(f'\n{self.replaylog[0]}\n', exc_info=e)
             # Discard and continue
             self.replaylog.pop(0)
+
         else:
             # Rewrite old schema name
             if msg['$schemaRef'].startswith('http://schemas.elite-markets.net/eddn/'):
-                msg['$schemaRef'] = 'https://eddn.edcd.io/schemas/' + msg['$schemaRef'][38:]
+                msg['$schemaRef'] = 'https://eddn.edcd.io/schemas/' + msg['$schemaRef'][38:]  # TODO: 38?!
+
```

Comment:
Indeed, replace with something actually referencing the string used in the conditional.

Although... we can either lost this code entirely *or* keep it as a reference on how to migrate older schema URLs.  It seems unlikely it would change *away* from eddn.edcd.io at this point though.